src/sparkx/Particle.py

```diff
@@ -364,15 +364,23 @@ def __initialize_from_array(self,input_format,particle_array):
                 "pz_": 6,
             },
         }
-        if (input_format in attribute_mapping) and (len(particle_array) == len(attribute_mapping[input_format])):
-            for attribute, index in attribute_mapping[input_format].items():
-                setattr(self, attribute, particle_array[index])
-
-            if input_format == "JETSCAPE":
-                self.mass_ = self.compute_mass_from_energy_momentum()
-                self.charge_ = self.compute_charge_from_pdg()
+        if (input_format in attribute_mapping):
+            if (len(particle_array) == len(attribute_mapping[input_format]) or (input_format in ["Oscar2013Extended","Oscar2013Extended_IC"]\
+                 and len(particle_array) <=  len(attribute_mapping[input_format])\
+                    and len(particle_array) >=  len(attribute_mapping[input_format])-2)):
+                for attribute, index in attribute_mapping[input_format].items():
+                    if len(particle_array)<index+1:
+                        setattr(self,attribute,None)
+                        continue
+                    setattr(self, attribute, particle_array[index])
```

Comment:
```suggestion
                    # Type casting for specific attributes
                    if attribute in ["t_", "x_", "y_", "z_", "mass_", "E_", "px_", "py_", "pz_", "form_time_", "xsecfac_", "t_last_coll_", "weight_"]:
                        setattr(self, attribute, float(particle_array[index]))
                    elif attribute in ["pdg_", "ID_", "charge_", "ncoll_", "proc_id_origin_", "proc_type_origin_", "pdg_mother1_", "pdg_mother2_", "baryon_number_", "strangeness_", "status_"]:
                        setattr(self, attribute, int(particle_array[index]))
                    else:
                        setattr(self, attribute, particle_array[index])
```

Reply:
I have added this part and checked that it works as expected.